### ISSUE TITLE
Issue if a CICS region is left on the GM screen for too long

### ISSUE THREAD

**Issue by hobbit1983** [galasa-dev/projectmanagement#313]
I have the following before class to define 2 terminals (one to use with CECI manager) and one to use for CEBR. I've left out the definition for the CECITerminal.
```
 @BeforeClass
    public void login()throws InterruptedException, Zos3270Exception {
        ...
        terminal.waitForKeyboard();
        terminal.type("logon applid(IYK2ZNB3)");
        terminal.enter();
        terminal.waitForTextInField("IYK2ZNB3");
        terminal.clear();
        terminal.waitForKeyboard();
}
```
later in the test class I have the following method which writes to a TSQ using CECI before:

```
 @Test
    public void writeToTSQTest()throws CECIException, InterruptedException, Zos3270Exception{
        String queueName = "queue1";
        String dataToWrite = "My name is Hobbit";
        String variableName = "TSQDATA";

        ceci.defineVariableText(ceciTerminal, variableName, dataToWrite);
        String command = "WRITEQ TS QUEUE('" + queueName +"') FROM(&" + variableName + ")";
        ceci.issueCommand(ceciTerminal, command);

        terminal.type("CEBR " + queueName).enter().waitForKeyboard();
        assertThat(terminal.retrieveScreen()).containsIgnoringCase(dataToWrite);
        terminal.pf3().waitForKeyboard().clear().waitForKeyboard();

        command = "DELETEQ TS QUEUE('" + queueName + "')";
        ceci.issueCommand(ceciTerminal, command);

        terminal.type("CEBR " + queueName).enter().waitForKeyboard();
        assertThat(terminal.retrieveScreen()).contains("DOES NOT EXIST");
        terminal.pf3().waitForKeyboard().clear().waitForKeyboard();        
    }
```
The test fails on the line `terminal.type("CEBR " + queueName).enter().waitForKeyboard();` with the error:

```
dev.galasa.zos3270.FieldNotFoundException: Unable to type where the cursor is pointing to - 0
        at dev.galasa.zos3270.spi.Screen.type(Screen.java:824)
        at dev.galasa.zos3270.spi.Screen.type(Screen.java:809)
        at dev.galasa.zos3270.spi.Terminal.type(Terminal.java:118)
        at dev.galasa.cicsts.ceci.manager.ivt.CECIManagerIVT.writeToTSQTest(CECIManagerIVT.java:191)
```

However if I add the following line to the end of my @BeforeClass then I can get the test to complete:
`terminal.type("CECI").enter().waitForKeyboard().pf3().clear();`
Which I don't quite understand as both with and without this line we should be at a clear screen.

I have the whole IVT in a branch CECIIVT off hobbit1983/managers if that helps

**Comment by Michael-Baylis** [comment]
yea. don't understand it either.   Will have a play

**Comment by hobbit1983** [comment]
Thanks. If you need trace or anything let me know

**Comment by Michael-Baylis** [comment]
Sorry Will,  I had to have a giggle when I figured this out.

In your beforeclass you issue a logon applid,     then you wait for the applid.   But it is so fast that the waitfortextinfield is finding the applid in your "logon applid(IYK2ZNB3)" text, therefore you are not in the CICS region when you think you are.

change your terminal.waitForTextInField("IYK2ZNB3"); 
to
terminal.waitForTextInField("DFHZC2312");
and it should work fine.
